static/js/giphy.js

```diff
@@ -0,0 +1,65 @@
+"use strict";
+
+const renderGrid = require("@giphy/js-components").renderGrid;
+const GiphyFetch = require("@giphy/js-fetch-api").GiphyFetch;
+const throttle = require("throttle-debounce").throttle;
+
+const gf = new GiphyFetch(page_params.giphy_api_key);
+let search_term = "trending";
+
+exports.renderGrid = function (targetEl) {
+    const render = () =>
+        renderGrid(
+            {
+                width: 300,
+                fetchGifs: (offset) =>
+                    gf.search(search_term, {
+                        offset,
+                        limit: 25,
+                        random_id: page_params.user_id,
+                    }),
+                columns: 3,
+                gutter: 6,
+                noLink: true,
+                hideAttribution: true,
+                onGifClick: (props) => {
+                    $("#compose_box_giphy_grid").popover("hide");
+                    $(
+                        "#compose-textarea",
+                    )[0].value += `[${search_term}](${props.images.downsized_medium.url})`;
```

Comment:
This should definitely use `compose_ui.insert_syntax_and_focus` -- that's our standard helper for adjusting the compose textarea.